src/error.rs

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![allow(clippy::module_name_repetitions)]
+
```

Comment:
Do we actually have module name repetitions in our code?

Reply:
Yes, for example, clippy counts: 
`pub enum CodecError<'a> {`
as a repetition of the word **error** in `error.rs`

I figured it is better to keep the names the same for now.

Reply:
Could update to `CodecErr` in the future